modules/calib3d/src/ptsetreg.cpp

```diff
@@ -501,6 +501,269 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
     }
 };
 
+class Affine2DEstimatorCallback : public PointSetRegistrator::Callback
+{
+public:
+    int runKernel( InputArray _m1, InputArray _m2, OutputArray _model ) const
+    {
+        Mat m1 = _m1.getMat(), m2 = _m2.getMat();
+        const Point2f* from = m1.ptr<Point2f>();
+        const Point2f* to   = m2.ptr<Point2f>();
+
+        // we need 3 points to estimate affine transform
+        const int N = 6;
+        double buf[N*N + N + N];
+        Mat A(N, N, CV_64F, &buf[0]);
+        Mat B(N, 1, CV_64F, &buf[0] + N*N);
+        Mat X(N, 1, CV_64F, &buf[0] + N*N + N);
+        double* Adata = A.ptr<double>();
+        double* Bdata = B.ptr<double>();
+        A = Scalar::all(0);
+
+        for( int i = 0; i < (N/2); i++ )
```

Comment:
Shouldn't runKernel() include all of the points in _m1 and _m2 in the estimate rather than just the first 3?  That's what HomographyEstimatorCallback.runKernel() does.  I presume that RANSAC will call this with sets of 3 points to determine the consensus set, but does it also call it at the end to refine the estimate with all of the points in the consensus set?